packages/flutter_tools/bin/xcode_backend.dart

```diff
@@ -0,0 +1,466 @@
+
+import 'dart:io';
+
+void main(List<String> arguments) {
+  Context(
+    arguments: arguments,
+    environment: Platform.environment,
+  ).run();
+}
+
+/// Container for script arguments and environment variables.
+///
+/// All interactions with the platform are broken into individual methods that
+/// can be overridden in tests.
+class Context {
+  Context({
+    required this.arguments,
+    required this.environment,
+  }) {
+    final String? scriptOutputStreamFileEnv = environment['SCRIPT_OUTPUT_STREAM_FILE'];
+    if (scriptOutputStreamFileEnv != null && scriptOutputStreamFileEnv.isNotEmpty) {
+      scriptOutputStreamFile = File(
+        scriptOutputStreamFileEnv,
+      ).openSync(mode: FileMode.write);
+    }
+  }
+
+  final Map<String, String> environment;
+  final List<String> arguments;
+
+  void run() {
+    if (arguments.isEmpty) {
+      // Named entry points were introduced in Flutter v0.0.7.
+      stderr.write(
+          'error: Your Xcode project is incompatible with this version of Flutter. '
+          'Run "rm -rf ios/Runner.xcodeproj" and "flutter create ." to regenerate.\n');
+      exit(-1);
+    }
+
+    final String subCommand = arguments.first;
+    switch (subCommand) {
+      case 'build':
+        buildApp();
+        break;
+      case 'thin':
+        // No-op, thinning is handled during the bundle asset assemble build target.
+        break;
+      case 'embed':
+        embedFlutterFrameworks();
+        break;
+      case 'embed_and_thin':
+        // Thinning is handled during the bundle asset assemble build target, so just embed.
+        embedFlutterFrameworks();
+        break;
+      case 'test_observatory_bonjour_service':
+        // Exposed for integration testing only.
+        addObservatoryBonjourService();
+    }
+  }
+
+  bool existsDir(String path) {
+    final Directory dir = Directory(path);
+    return dir.existsSync();
+  }
+
+  bool existsFile(String path) {
+    final File file = File(path);
+    return file.existsSync();
+  }
+
+  /// Run given command in a synchronous subprocess.
+  ///
+  /// Will throw [Exception] if the exit code is not 0.
+  ProcessResult runSync(
+    String bin,
+    List<String> args, {
+    bool verbose = false,
+    bool allowFail = false,
+    String? workingDirectory,
+  }) {
+    if (verbose) {
+      print('♦ $bin ${args.join(' ')}');
+    }
+    final ProcessResult result = Process.runSync(
+      bin,
+      args,
+      workingDirectory: workingDirectory,
+    );
+    echo(result.stdout as String);
+    if ((result.stderr as String).isNotEmpty) {
+      echoError(result.stderr as String);
+    }
+    if (!allowFail && result.exitCode != 0) {
+      stderr.write('${result.stderr}\n');
+      throw Exception(
+        'Command "$bin ${args.join(' ')}" exited with code ${result.exitCode}',
+      );
+    }
+    return result;
+  }
+
+  /// Log message to stderr.
+  void echoError(String message) {
+    stderr.write('$message\n');
```

Comment:
If you need a newline:
```suggestion
    stderr.writeln(message);
```

Reply:
done